projects/plugins/jetpack/extensions/blocks/subscriptions/class-jetpack-subscription-site.php

```diff
@@ -39,6 +40,27 @@ public static function init() {
 	 */
 	public function handle_subscribe_block_placements() {
 		$this->handle_subscribe_block_post_end_placement();
+
+		add_filter(
+			'jetpack_options_whitelist',
+			function ( $options ) {
+				$options[] = 'jetpack_subscriptions_subscribe_navigation_enabled';
+
+				return $options;
+			}
+		);
```

Comment:
Let's remove it from here and add the option to the `options` array there:
https://github.com/Automattic/jetpack/blob/75b03102a816848ca9a6c0ca084887b6237a3f66/projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php#L176-L183